cli.js

```diff
@@ -112,17 +117,27 @@ api.on('error', logger.unhandledError);
 api.on('stdout', logger.stdout);
 api.on('stderr', logger.stderr);
 
-api.run()
-	.then(function () {
-		logger.finish();
-		logger.exit(api.failCount > 0 || api.rejectionCount > 0 || api.exceptionCount > 0 ? 1 : 0);
-	})
-	.catch(function (err) {
-		if (err.name === 'AvaError') {
-			console.log('  ' + colors.error(figures.cross) + ' ' + err.message);
-		} else {
-			console.error(colors.stack(err.stack));
-		}
-
-		logger.exit(1);
+function run() {
+	api.run()
+		.then(function () {
+			logger.finish();
+			logger.exit(api.failCount > 0 || api.rejectionCount > 0 || api.exceptionCount > 0 ? 1 : 0);
+		})
+		.catch(function (err) {
+			if (err.name === 'AvaError') {
+				console.log('  ' + colors.error(figures.cross) + ' ' + err.message);
+			} else {
+				console.error(colors.stack(err.stack));
+			}
+
+			logger.exit(1);
 	});
+}
+
+run();
+
+var watcher = chokidar.watch(cli.input, {persistent: cli.flags.watch});
```

Comment:
I was thinking whether it would be more effective watch the whole `cwd` instead of just test files, but wasn't sure what kind of performance impact would that have.